user-interface/src/main/java/life/qbic/datamanager/views/projects/project/samples/SampleDetailsComponent.java

```diff
@@ -130,6 +145,57 @@ private void initButtonAndFieldBar() {
     buttonAndFieldBar.addClassName("button-and-search-bar");
   }
 
+  private void onRegisterButtonClicked(ClickEvent<Button> clickEvent) {
+    Experiment experiment = context.experimentId()
+        .flatMap(experimentInformationService::find)
+        .orElseThrow();
+
+    BatchRegistrationDialog2 dialog = new BatchRegistrationDialog2(
+        experiment.getName(), new ArrayList<>(experiment.getSpecies()),
+        new ArrayList<>(experiment.getSpecimens()), new ArrayList<>(experiment.getAnalytes()),
+        experiment.getExperimentalGroups());
+
+    dialog.addCancelListener(cancelEvent -> cancelEvent.getSource().close());
+    dialog.addConfirmListener(this::onRegisterDialogConfirmed);
+    dialog.open();
+  }
+
+  private void onRegisterDialogConfirmed(ConfirmEvent confirmEvent) {
+    Data data = confirmEvent.getData();
```

Comment:
I think the readability could be improved by not getting "data" but something more explicit like "BatchRegistrationContent"
```suggestion
    Data data = confirmEvent.getData();
```